stripe/functions/index.js

```diff
@@ -18,77 +18,142 @@
 const functions = require('firebase-functions');
 const admin = require('firebase-admin');
 admin.initializeApp();
-const logging = require('@google-cloud/logging')();
-const stripe = require('stripe')(functions.config().stripe.token);
-const currency = functions.config().stripe.currency || 'USD';
-
-// [START chargecustomer]
-// Charge the Stripe customer whenever an amount is created in Cloud Firestore
-exports.createStripeCharge = functions.firestore.document('stripe_customers/{userId}/charges/{id}').onCreate(async (snap, context) => {
-      const val = snap.data();
-      try {
-        // Look up the Stripe customer id written in createStripeCustomer
-        const snapshot = await admin.firestore().collection(`stripe_customers`).doc(context.params.userId).get()
-        const snapval = snapshot.data();
-        const customer = snapval.customer_id
-        // Create a charge using the pushId as the idempotency key
-        // protecting against double charges
-        const amount = val.amount;
-        const idempotencyKey = context.params.id;
-        const charge = {amount, currency, customer};
-        if (val.source !== null) {
-          charge.source = val.source;
-        }
-        const response = await stripe.charges.create(charge, {idempotency_key: idempotencyKey});
-        // If the result is successful, write it back to the database
-        return snap.ref.set(response, { merge: true });
-      } catch(error) {
-        // We want to capture errors and render them in a user-friendly way, while
-        // still logging an exception with StackDriver
-        console.log(error);
-        await snap.ref.set({error: userFacingMessage(error)}, { merge: true });
-        return reportError(error, {user: context.params.userId});
-      }
-    });
-// [END chargecustomer]]
+const { Logging } = require('@google-cloud/logging');
+const logging = new Logging({
+  projectId: '<YOUR FIREBASE PROJECT ID>',
```

Comment:
Instead of asking a developer to set this manually, this can be accessed as an environment variable that's automatically provided to Cloud Functions via `process.env.GCLOUD_PROJECT`